mongo/datadog_checks/mongo/collectors/custom_queries.py

```diff
@@ -202,6 +203,8 @@ def collect(self, api):
         for raw_query in self.custom_queries:
             try:
                 self._collect_custom_metrics_for_query(api, raw_query)
+            except CRITICAL_FAILURE as e:
+                raise e  # Critical failures must bubble up to trigger a CRITICAL service check.
```

Comment:
Is the exact error logged somewhere else? I don't see a log in the `check` method

Reply:
Ah, this is a good point.  [here](https://github.com/DataDog/integrations-core/pull/15208#discussion_r1284350675) Alex proposes to reraise the exception so that it gets handled by the global dispatch in the `run` method. I'm leaning in that direction too tbh, since I'd like us to long-term centralize most exception-handling there.

Reply:
It turns out we rely on the exception not being raised, tests fail. I could add a logging line to match the general exception case.

Reply:
Updated tests, resolving this thread.